### DISCUSSION TITLE
Is it possible to inject a command while i proxy-mode?

### DISCUSSION THREAD

**Discussion by easyjoh** [johanmeijer/grott#365] [Q&A]
Hi,

I want to be able to limit export based on rules, e.g. via home-assistant automations.

Still I would prefer to forward data to server.grott.com and also be able to set other command via the shine-app.

So is it possible to inject a command to enable/disable the export limit (register 122) in that case?

**Comment by johanmeijer** [comment]
No not possible at this moment. In the future i want to integrate grott and grottserver. Then it want to make this possible.

**Reply by easyjoh** [reply]
Ok. Looking forward to that.

**Reply by fhteagle** [reply]
Also looking forward to this, especially as a feature of grottserver. Any idea if it will be possible to command inverter output on/off/wakeup after powersaving timeout? If not I probably can rig a raspberry pi with a relay to automate the physical switch "toggle" electrically....

**Comment by rany2** [comment]
For the time being, you might want to use my fork at https://github.com/rany2/grott which supports grottserver + forwarding to growatt's servers at the same time which means you could inject commands while preserving data forwarding to server.growatt.com. After you switch to it look into https://github.com/rany2/grott/blob/master/examples/grott.ini  and set `serverforward` to True and `mode` to server. (environs would be `gserverforward` to True and `gmode` to `server`)

**Reply by easyjoh** [reply]
Nice. I will look into it and give it a try.
Have you considered a pull request to merge into the origin? Could save @johanmeijer some time

**Reply by rany2** [reply]
I am planning to do so, but the issue is the pull request would be too massive as I've diverged quite massively from johan's grott. I'd need to compartmentalize it which isn't as easy as it sounds
